packages/flutter_tools/test/general.shard/ios/fallback_discovery_test.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/ios/fallback_discovery.dart';
+import 'package:flutter_tools/src/mdns_discovery.dart';
+import 'package:flutter_tools/src/protocol_discovery.dart';
+import 'package:mockito/mockito.dart';
+import 'package:platform/platform.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+import '../../src/mocks.dart';
+
+// This test still uses `testUsingContext` due to analytics usage.
+void main() {
+  BufferLogger logger;
+  FallbackDiscovery fallbackDiscovery;
+  MockMDnsObservatoryDiscovery mockMDnsObservatoryDiscovery;
+  MockPrototcolDiscovery mockPrototcolDiscovery;
+  MockPortForwarder mockPortForwarder;
+  MockVmService mockVmService;
+
+  setUp(() {
+    logger = BufferLogger(
+      terminal: AnsiTerminal(stdio: MockStdio(), platform: const LocalPlatform()),
+      outputPreferences: OutputPreferences.test(),
+    );
+    mockVmService = MockVmService();
+    mockMDnsObservatoryDiscovery = MockMDnsObservatoryDiscovery();
+    mockPrototcolDiscovery = MockPrototcolDiscovery();
+    mockPortForwarder = MockPortForwarder();
+    fallbackDiscovery = FallbackDiscovery(
+      logger: logger,
+      mDnsObservatoryDiscovery: mockMDnsObservatoryDiscovery,
+      portForwarder: mockPortForwarder,
+      protocolDiscovery: mockPrototcolDiscovery,
+      vmServiceConnectUri: (String uri, {Log log}) async {
+        return mockVmService;
+      },
+    );
+    when(mockPortForwarder.forward(23, hostPort: anyNamed('hostPort')))
+      .thenAnswer((Invocation invocation) async => 1);
+  });
+
+  testUsingContext('Selects assumed port if VM service connection is successful', () async {
```

Comment:
Sorry if I already asked this, but can these be `testWithoutContext()` ?

Reply:
I think you mentioned it and I completely forgot

Reply:
Ahh, no it was the `Usage/event` classes need to be refactored. There doesn't seem to be a way to reference any of the helpers without going through the context

Reply:
Ah, okay. I'll take on the `Usage` refactor when I'm back in the office.